packages/outline-core-link/src/outline-core-link.ts

```diff
@@ -85,20 +95,102 @@ export class OutlineCoreLink extends OutlineElement {
     return this.isURLExternal(this.linkHref) ? '_blank' : undefined;
   }
 
-  render(): TemplateResult {
+  generateLink(): TemplateResult {
+    return html` <a
+      href=${this.linkHref}
+      rel="${ifDefined(this.linkRelRender())}"
+      target="${ifDefined(this.linkTargetRender())}"
+    >
+      ${this.linkText ? html`${this.linkText}` : html`<slot></slot>`}
+    </a>`;
+  }
+
+  /**
+   * Get all elements in the default slot.
+   * @todo - Move method to a controller.
+   *
+   * @returns NodeList of all elements in the default slot.
+   */
+  getSlottedContent(): NodeList {
+    return this.querySelectorAll('*');
+  }
+
+  /**
+   * Check to see if the element is slotted properly.
+   * @todo - Move method to a controller.
+   *
+   * @returns boolean
+   */
+  isValidTopLevelSlottedLink(): boolean {
+    const slot: NodeList = this.getSlottedContent();
+    if (slot.length === 1 && slot[0].nodeName === 'A') {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * If the element is not slotted properly, log an error to the console.
+   * @todo - Enable a global debug mode in outline.config.js that will determine if the console.group is logged in this and other component level debugging code.
+   * @todo - Implement a controller to handle debugging Outline components.
+   *
+   * @returns void
+   */
+  debugSlottedContent(): void {
+    console.group(componentName);
+    console.error(
+      `${componentName} must have a single <a> tag as a child of the default slot.`
+    );
+    console.log(this.getSlottedContent());
+    console.groupEnd();
+  }
+
+  /**
+   * Adjust attributes on a slotted link.
+   *
+   * When the link is slotted, we'll test, modify & render the slotted link,
+   * and adjust the target and rel attributes if they are set on the outline-core-link element.
+   */
+  adjustSlottedContent(): void {
+    const slottedLink: HTMLAnchorElement | null = this.querySelector('a');
+    if (this.linkTarget) {
+      slottedLink?.setAttribute('target', this.linkTarget);
+    }
+    if (this.linkRel) {
+      slottedLink?.setAttribute('rel', this.linkRel);
+    }
+  }
+
+  /**
+   * If the element is fully slotted, return the full markup in the default slot.
+   *
+   * @todo - Enable a global debug mode in outline.config.js that will determine if the console.group is logged.
+   * @returns HTMLSlotElement
+   */
+  fullMarkupInSlot(): TemplateResult {
+    const debug = false;
+    if (!this.isValidTopLevelSlottedLink()) {
+      if (debug) this.debugSlottedContent();
```

Comment:
I think only `if (debug)` we should call `isValidTopLevelSlottedLink()`

Reply:
Good catch. No need for the logic going on there. Other comments made, I'll adjust this last bit.

Reply:
I've made this so that the logic is more straightforward, and until the "debugController` is in place, I want to see this console error in the browser if someone is implementing the component improperly. So the debug mode is "on" always in this case.